addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6689,23 +6773,24 @@ registry.ImageTools = ImageHandlerOption.extend({
                         img.classList.remove("o_animate_on_hover");
                     }
                     if (!this._isAnimatedShape()) {
-                        delete img.dataset.shapeAnimationSpeed;
+                        delete this.imageOptions.shapeAnimationSpeed;
                     }
                 }
+                await this._applyShapeAndColors(saveData);
             }
         } else {
             // Re-applying the modifications and deleting the shapes
-            img.src = await applyModifications(img, {mimetype: this._getImageMimetype(img)});
-            delete img.dataset.shape;
-            delete img.dataset.shapeColors;
-            delete img.dataset.fileName;
-            delete img.dataset.shapeFlip;
-            delete img.dataset.shapeRotate;
-            delete img.dataset.shapeAnimationSpeed;
+            delete this.imageOptions.shape;
+            delete this.imageOptions.shapeColors;
+            delete this.imageOptions.fileName;
+            delete this.imageOptions.shapeFlip;
+            delete this.imageOptions.shapeRotate;
+            delete this.imageOptions.shapeAnimationSpeed;
             if (saveData) {
-                img.dataset.mimetype = img.dataset.originalMimetype;
-                delete img.dataset.originalMimetype;
+                this.imageOptions.mimetype = this.imageOptions.originalMimetype;
+                delete this.imageOptions.originalMimetype;
             }
+            await this._applyOptions();
```

Comment:
Taking the advantage of the `_applyOptions()` method to remove `img.src = await applyModifications(img, {mimetype: this._getImageMimetype(img)});` and to correctly update the image options in the registry at the end of the `imageOptions` modification.